Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -1425,24 +1477,23 @@ internal struct JSONScanner {
   // they don't know; newer clients may reject the same input due to
   // schema mismatches or other issues.
   private mutating func skipString() throws {
-    if currentByte != asciiDoubleQuote {
-      throw JSONDecodingError.malformedString
+    guard hasMoreContent && currentByte == asciiDoubleQuote else {
+      throw JSONDecodingError.truncated
```

Comment:
.truncated seems correct if hasMoreContent failed, but if we really have to worry about coming in on something other than a quote, it seems like .malformedString is more correct.  Is the currentByte check more of an assert/precondition that isn't really needed in production code?

Reply:
Good point.  I've split this into two separate checks that throw separate errors.